sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -744,3 +748,43 @@ def test_single_node_trees(Est):
 def test_custom_loss(Est, loss, X, y):
     est = Est(loss=loss, max_iter=20)
     est.fit(X, y)
+
+
+@pytest.mark.parametrize('HistGradientBoosting, X, y', [
+    (HistGradientBoostingClassifier, X_classification, y_classification),
+    (HistGradientBoostingRegressor, X_regression, y_regression),
+    (HistGradientBoostingClassifier,
+        X_multi_classification, y_multi_classification),
+])
+def test_staged_predict(HistGradientBoosting, X, y):
+
+    # Test whether staged predictor eventually gives
+    # the same prediction.
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y,
+        test_size=0.5,
+        random_state=0
+    )
+    gb = HistGradientBoosting(
+        max_iter=10,
+        random_state=0
+    )
+
+    # test raise NotFittedError if not fitted
+    with pytest.raises(NotFittedError):
+        next(gb.staged_predict(X_test))
+
+    gb.fit(X_train, y_train)
+    y_pred = gb.predict(X_test)
+
+    # test if the staged predictions of each iteration
+    # are equal to the corresponding predictions of the same estimator
+    # trained from scratch
+    # this also test limit case when max_iter = 1
+    for n_iter, y in enumerate(gb.staged_predict(X_test), 1):
```

Comment:
(for some reason this comment didn't go through so I'm reposting)

let's test all methods here. e.g.

```py
from sklearn.base import is_regressor  # at the top

method_names = (['predict'] if is_regressor(gb) else ['predict', 'predict_proba', 'decision_function']

for method_named in method:
	method = getattr(...)
	for iter in ...
		staged_method = getatr(aux, 'staged_' + method_name)
		...
```